apps/storage/api/avatar.post.ts

```diff
@@ -14,192 +14,191 @@ import { PutObjectCommand } from '@aws-sdk/client-s3';
+                where: eq(orgMembers.role, 'admin')
+              }
             }
           }
         }
+      });
+      if (!groupResponse) {
+        setResponseStatus(event, 400);
+        return send(event, 'Invalid group');
+      }
+      const isAdmin = groupResponse.org.members.some(
+        (member) => +member.userId === +userId
+      );
+      if (!isAdmin) {
+        setResponseStatus(event, 401);
+        return send(event, 'Unauthorized');
       }
-    });
-    if (!groupResponse) {
+    } else {
       setResponseStatus(event, 400);
-      return send(event, 'Invalid group');
+      return send(event, 'Invalid type value');
     }
-    const isAdmin = groupResponse.org.members.some(
-      (member) => +member.userId === +userId
-    );
-    if (!isAdmin) {
-      setResponseStatus(event, 401);
-      return send(event, 'Unauthorized');
+
+    const file = formInputs.find((input) => input.name === 'file');
+    if (!file) {
+      setResponseStatus(event, 400);
+      return send(event, 'Missing file attachment');
     }
-  } else {
-    setResponseStatus(event, 400);
-    return send(event, 'Invalid type value');
-  }
+    const acceptedFileTypes = ['image/jpeg', 'image/png', 'image/gif'];
+    if (!acceptedFileTypes.includes(file.type)) {
+      setResponseStatus(event, 400);
+      return send(event, 'Invalid file type');
+    }
+    // do image conversion
+    const sizes = [
+      { name: '3xs', value: 16 },
+      { name: '2xs', value: 20 },
+      { name: 'xs', value: 24 },
+      { name: 'sm', value: 32 },
+      { name: 'md', value: 40 },
+      { name: 'lg', value: 48 },
+      { name: 'xl', value: 56 },
+      { name: '2xl', value: 64 },
+      { name: '3xl', value: 80 },
+      { name: '4xl', value: 96 },
+      { name: '5xl', value: 128 }
+    ];
 
-  const file = formInputs.find((input) => input.name === 'file');
-  if (!file) {
-    setResponseStatus(event, 400);
-    return send(event, 'Missing file attachment');
-  }
-  const acceptedFileTypes = ['image/jpeg', 'image/png', 'image/gif'];
-  if (!acceptedFileTypes.includes(file.type)) {
-    setResponseStatus(event, 400);
-    return send(event, 'Invalid file type');
-  }
-  // do image conversion
-  const sizes = [
-    { name: '3xs', value: 16 },
-    { name: '2xs', value: 20 },
-    { name: 'xs', value: 24 },
-    { name: 'sm', value: 32 },
-    { name: 'md', value: 40 },
-    { name: 'lg', value: 48 },
-    { name: 'xl', value: 56 },
-    { name: '2xl', value: 64 },
-    { name: '3xl', value: 80 },
-    { name: '4xl', value: 96 },
-    { name: '5xl', value: 128 }
-  ];
+    for (const size of sizes) {
+      const resizedImage = await sharp(file.data)
+        .resize(size.value, size.value)
+        .toBuffer();
+      const command = new PutObjectCommand({
+        Bucket: 'avatars',
+        Key: `${typeObject.value}_${avatarId}/${size.name}`,
+        Body: resizedImage,
+        ContentType: file.type
+      });
+      await s3Client.send(command);
```

Comment:
This will load the image into memory 11 times. If the original image is big (like 25mb) this will be > 250mb of ram :o

Reply:
ummm....
thats a good point, didnt realize it did that

we could open another issue for this, since this change was just formating/linting and not the actual implementation